HabboHotel/Groups/Forums/GroupForumThread.cs

```diff
@@ -0,0 +1,188 @@
+        {
+            _database = database;
+            Views = new List<GroupForumThreadPostView>(); ;
+            UsersOnThread = new List<GameClient>();
+            ParentForum = parent;
+
+            Id = id;
+            UserId = userid;
+            Timestamp = timestamp;
+            Caption = caption;
+            Posts = new List<GroupForumThreadPost>();
+
+            Pinned = pinned;
+            Locked = locked;
+            DeletedLevel = deletedlevel;
+            DeleterUserId = deleterid;
+            DeletedTimestamp = (int)PlusEnvironment.GetUnixTimestamp();
+
+            using var connection = _database.Connection();
+            var results = connection.Query("SELECT * FROM group_forums_thread_posts WHERE thread_id = @id", new { id = Id });
+
+            foreach (var result in results)
+            {
+                Posts.Add(new GroupForumThreadPost(_database, this, Convert.ToInt32(result.id), Convert.ToInt32(result.user_id), Convert.ToInt32(result.timestamp), result.message.ToString(), Convert.ToInt32(result.deleted_level), Convert.ToInt32(result.deleter_user_id)));
+            }
+
+            var results2 = connection.Query("SELECT * FROM group_forums_thread_views WHERE thread_id = @id", new { id = Id });
+
+            foreach (var result2 in results2)
+            {
+                Views.Add(new GroupForumThreadPostView(Convert.ToInt32(result2.id), Convert.ToInt32(result2.userid), Convert.ToInt32(result2.count)));
+            }
+        }
+
+        public async void AddView(int userid, int count = -1)
+        {
+            GroupForumThreadPostView view;
+            var now = (int)PlusEnvironment.GetUnixTimestamp();
+
+            if ((view = GetView(userid)) != null)
+            {
+                view.Count = count >= 0 ? count : Posts.Count;
+                using var connection = _database.Connection();
+                await connection.ExecuteAsync("UPDATE group_forums_thread_views SET count = @c, timestamp = @timestamp WHERE thread_id = @p AND user_id = @u", new { c = view.Count, timestamp = now, p = Id, u = userid });
+            }
+            else
+            {
+                view = new GroupForumThreadPostView(0, userid, Posts.Count);
+
+                using var connection = _database.Connection();
+                var sqlStatement = @"INSERT INTO group_forums_thread_views (thread_id, timestamp, user_id, count) VALUES (@threadId, @timestamp, @userId, @count); SELECT LAST_INSERT_ID()";
+
+                view.Id = await connection.ExecuteScalarAsync<int>(sqlStatement, new
+                {
+                    threadId = Id,
+                    timestamp = now,
+                    userId = userid,
+                    count = view.Count
+                });
+
+                Views.Add(view);
+            }
+        }
+
+        public GroupForumThreadPostView GetView(int userid) => Views.FirstOrDefault(count => count.UserId == userid);
+
+        public int GetUnreadMessages(int userid)
```

Comment:
I dont understand this method. What if I call it with two different user ids, then each view the userid is overwritten?